connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java

```diff
@@ -77,6 +81,8 @@
     private HashMap<ConnectorTaskId, WorkerTask> tasks = new HashMap<>();
     private KafkaProducer<byte[], byte[]> producer;
     private SourceTaskOffsetCommitter sourceTaskOffsetCommitter;
+    private static final List<Class<? extends Connector>>
+        SKIPPED_CONNECTORS = Arrays.asList(SourceConnector.class, SinkConnector.class, VerifiableSourceConnector.class, VerifiableSinkConnector.class);
```

Comment:
Special casing some of these makes sense (e.g. the `Verifiable*Connector` classes), but others are skipped for a different reason: they are interfaces/abstract classes. Can we handle these more generally by checking the `Class.getModifiers()` for each class maybe? Ideally we could filter out abstract classes added by connector developers as well (which they might have if they can share code between a few different connectors).